modules/ding_permissions/ding_permissions.module

```diff
@@ -1000,12 +1001,10 @@ function ding_permissions_secure_permissions($role) {
  * Implements hook_preprocess_HOOK().
  */
 function ding_permissions_preprocess_html(&$variables) {
-  global $user;
```

Comment:
Good idea. Didn't think of putting the logic in secure permissions hook.

So if the role has `access administration pages` we make sure it has admin_menu and not toolbar.

We could also make sure that a role without `access administration pages` doesn't have permissions to both. What do you think?